lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -96,15 +110,40 @@ sub _parse_obs_response_dirty {
     my ($res) = @_;
 
     my $results = $res->dom('result');
-    return undef unless $results->size;
+    return (undef, '') unless $results->size;
 
     for my $result ($results->each) {
         my $attributes = $result->attr;
-        return 1 if exists $attributes->{dirty};
+        return (1, 'dirty') if exists $attributes->{dirty};
         next if ($attributes->{repository} // '') ne 'images';
-        return 1 if ($attributes->{state} // '') ne 'published';
+        return (1, $attributes->{state} // '') if ($attributes->{state} // '') ne 'published';
     }
-    return 0;
+    return (0, 'published');
+}
+
+# This method is coupled with openqa-trigger-from-obs and returns
+# string in format %y%m%d_%H%M%S, which corresponds to location
+# used by openqa-trigger-from-obs to determine if any files changed
+# or rsync can be skipped.
+sub _get_run_last_info {
+    my ($c, $project) = @_;
+    my $home = $c->obs_rsync->home;
+    my $run;
+    eval { $run = readlink(Mojo::File->new($home, $project, '.run_last')) };
+    return "" unless $run;
+    my $timestring = Mojo::File->new($run)->basename;
+    $timestring =~ s/^.run_//;
+    return $timestring;
+}
+
+# Dirty status file is updated from ObsRsync Gru tasks
+sub _get_dirty_status {
+    my ($c, $project) = @_;
+    my $home = $c->obs_rsync->home;
+    open(my $fh, '<', Mojo::File->new($home, $project, $dirty_status_filename)) or return "";
```

Comment:
should you really silently accept that open would not work? How about using autodie?

Reply:
yes, the plan is that visual enhancements in this PR do its best to display additional informational values and just ignore eventual errors.

Reply:
I can also recommend to use `autodie` to keep the code clean

Reply:
No, most of IO operations here are for cosmetic needs and their failure must not disturb main logic (e.g. in obs_rsync.is_status_dirty helper, which also caches last status for UI)